src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2229,4 +2231,18 @@ public final XmlFile getStateFile() {
             return new XmlFile(Items.XSTREAM, new File(owner.getRootDir(), "state.xml"));
         }
     }
+
+    /**
+     * Veto attempts to copy branch projects outside of their multibranch container. Only works on Jenkins core
+     * versions with JENKINS-34691 merged.
```

Comment:
Good to have an integration test, `@Ignore`d prior to the version introducing the API.